src/common/post.ts

```diff
@@ -252,3 +257,80 @@ export const validatePost = (
 export const submitArticleThreshold = parseInt(
   process.env.SUBMIT_ARTICLE_THRESHOLD,
 );
+
+export const insertCodeSnippets = async ({
+  entityManager,
+  post,
+  codeSnippetsJson,
+}: {
+  entityManager: EntityManager;
+  post: Pick<Post, 'id'>;
+  codeSnippetsJson: PostCodeSnippetJsonFile;
+}) => {
+  const codeSnippets = codeSnippetsJson.snippets.reduce(
+    (acc, codeSnippetContent, index) => {
+      const checksum = createHash('sha1');
+      checksum.update(codeSnippetContent);
+      const contentHash = checksum.digest('hex');
+
+      if (!acc.has(contentHash)) {
+        acc.set(
+          contentHash,
+          entityManager.getRepository(PostCodeSnippet).create({
+            postId: post.id,
+            contentHash,
+            order: index,
+            content: codeSnippetContent,
+          }),
+        );
+      }
+
+      return acc;
+    },
+    new Map<string, PostCodeSnippet>(),
+  );
+
+  const uniqueCodeSnippets = Array.from(codeSnippets.values());
+
+  await entityManager.getRepository(PostCodeSnippet).delete({
+    postId: post.id,
+    contentHash: Not(
+      In(uniqueCodeSnippets.map((snippet) => snippet.contentHash)),
+    ),
+  });
+
+  await entityManager
+    .getRepository(PostCodeSnippet)
+    .upsert(uniqueCodeSnippets, {
+      conflictPaths: ['postId', 'contentHash'],
+    });
+};
+
+export const insertCodeSnippetsFromUrl = async ({
+  entityManager,
+  post,
+  codeSnippetsUrl,
+}: {
+  entityManager: EntityManager;
+  post: Pick<Post, 'id'>;
+  codeSnippetsUrl: string | undefined;
+}) => {
+  try {
+    if (!codeSnippetsUrl) {
+      return;
+    }
+
+    const codeSnippetsJson = await downloadJsonFile<PostCodeSnippetJsonFile>({
+      url: codeSnippetsUrl,
```

Comment:
It's one url or can be multiple?
Mainly asking because the other part is called:
`data?.meta?.stored_code_snippets,` (sounds like it could be more than one)

Reply:
yes, only one URL pointing to file that contains all code snippets, I was confused as well but confirmed with Denis.